csharp/src/Ice/OutputStream.cs

```diff
@@ -1651,9 +1522,39 @@ public void WriteEnum(int tag, int? v, int maxValue)
         /// <param name="v">The value to write.</param>
         public void WriteClass(AnyClass? v)
         {
-            initEncaps();
-            Debug.Assert(_encapsStack != null && _encapsStack.encoder != null);
-            _encapsStack.encoder.WriteClass(v);
+            Debug.Assert(_mainEncaps != null && _endpointEncaps == null);
+            if (v == null)
+            {
+                WriteSize(0);
+            }
+            else if (_current != null && _format == FormatType.SlicedFormat)
+            {
+                _current.IndirectionTable ??= new List<AnyClass>();
+                _current.IndirectionMap ??= new Dictionary<AnyClass, int>();
+
+                // If writing an instance within a slice and using the sliced format, write an index of that slice's
+                // indirection table.
+                int index = 0;
+                if (_current.IndirectionMap != null && _current.IndirectionMap.TryGetValue(v, out index))
```

Comment:
Why do we check for null here?

Reply:
The redundant lazy initialization above is incorrect. Will fix.